ARCs/arc-0004.md

```diff
@@ -527,12 +573,21 @@ any leading zeros, _unless_ `N` is zero, in which case only a single 0 character
 * `<type>[]`: A variable-length array. `type` can be any other type.
 * `string`: A variable-length byte array (`byte[]`) assumed to contain UTF-8 encoded content.
 * `(T1,T2,…,TN)`: A tuple of the types `T1`, `T2`, …, `TN`, `N >= 1`.
+* `(name1:T1,name2:T2,…,nameN:TN)`: A named tuple of the types `T1`, `T2`, …, `TN`, `N >= 1`, for use in contract.json to help representation of structs/objects from HLLs.
 * reference types `account`, `asset`, `application`: **MUST NOT** be used as the return type.
 For encoding purposes they are an alias for `uint8`. See section "Reference Types" below.
 
 Additional special use types are defined in [Reference Types](#reference-types)
 and [Transaction Types](#transaction-types).
 
+### Type hints
+The following hints to further describe an argument or return type are available. 
+Hints are arrays of functions in an argument description:
+
+* Max(N) - the maximum decimal or hexadecimal ('0x...') value,N, this integer can contain
+* Min(N) - the minimum decimal or hexadecimal ('0x...') value,N, this integer can contain
+* ...
+
```

Comment:
I am v hesitant to add these as any kind of mini DSL for hints, id rather just put the hint in text in the description

Reply:
I agree. The ABI should only describe how to encode a value, not information about the semantics of the values. For example, maybe a method only takes odd numbers. That information should not be encoded in the ABI.

Reply:
Yes. This information should belong in the Arc24 extension to arc4 that serves for client generarion. Aliases can be established. For example uint16 witha Precision hint of a positive value works far better than a FixedNxM value. By the way disagree on semantics. Arc4 specifies a fixed point arithmetic type but *should* mandate santics around the type